src/test/kotlin/sh/hsp/techtree/YamlReaderTest.kt

```diff
@@ -2,28 +2,25 @@ package sh.hsp.techtree
 
 import io.kotest.matchers.equals.shouldBeEqual
 import org.junit.jupiter.api.Test
+import java.net.URL
 
 class YamlReaderTest {
-    val yamlReader: YamlReader? = null;
-
+    private val yamlReader: YamlReader = YamlReaderImplemantion()
```

Comment:
Technicznie mógłbyś zrobić tak:
```suggestion
    private val yamlReader: YamlReader = object : YamlReader {}
```
Wtedy nie musiałbyś deklarować tego `YamlReaderImplemantion()`, nie wiem czy to jest taki odpowiednik javowy czegoś takiego: 
```
public class Clazz {
    public static void example() {
        Interface x = new Interface() {};
    }
}
```

Natomiast tak czy inaczej - raczej nie tędy droga

See: https://github.com/hspsh/tech-tree/pull/2#discussion_r1480351759